### PR TITLE
React redux/feature

### PR DESCRIPTION
**Pull request by Haadiiii**
## Microverse Redux in React components Project ⏬ 

in this project :

- Imported Redux Provider and stored in the main component.
- Connect the app to the Redux store
- Used use selectors to the state for `Display Book`

> DIsplay Book

-  Added new books, add functionality for the "Add new" button click.
- imported Redux useDispatch.
- generate a unique id and add them to an object.
- dispatch a corresponding action.

> Remove button

- Added removing books, added a functionality.
- import Redux useDispatch.
- Dispatched a corresponding action

> Check status Button

- added a functionality
- Import Redux useDispatch.
- Dispatched a corresponding action

👍  😄

### CODE REVIEW

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @Haadiiii , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights

- PR has a good title and summary :heavy_check_mark: 
- PR has a correct GitHub workflow :heavy_check_mark: 
- No Linter error :heavy_check_mark: 

## Required Changes ♻️

- [ ] Hi @Haadiiii, please thoroughly customize all the headers in the README file to match your project. You just paste the template and didn't customize the content at all :angry: 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: package.json [hunk 1]
```diff
@@ -13,7 +13,9 @@
     "react": "^18.2.0",
     "react-dom": "^18.2.0",
     "react-redux": "^8.0.2",
-    "react-router-dom": "^6.4.0"
+    "react-router-dom": "^6.4.0",
+    "react-uuid": "^1.0.3",
+    "redux": "^4.2.0"
   },
   "devDependencies": {
     "@babel/core": "^7.19.1",
```
[the remark this answers is not in the document]
Comment (Haadiiii) [reply package.json:21]: you should use npm start or npm run dev its working fine by me 👍  see screenshot

image

File: src/Components/Book.jsx [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+        title:
+        {props.title}
+      </div>
+      <div>
+        author:
+        {props.author}
+      </div>
+      <button type="button" onClick={() => dispatch(removeBook(props.id))}>
+        Remove
+      </button>
```
[the remark this answers is not in the document]
Comment (Haadiiii) [reply src/Components/Book.jsx:19]: thanks ill do that 👍

File: src/pages/Categories.jsx [hunk 3]
```diff
@@ -1,9 +1,16 @@
...
-  <div className="cati">
-    <button type="button">Check status</button>
-  </div>
-);
+const Categories = () => {
+  const dispatch = useDispatch();
+  const { message } = useSelector((state) => state.category);
+  return (
+    <div className="cati">
+      <button type="button" onClick={() => dispatch(checkBook())}>Check status</button>
```
[the remark this answers is not in the document]
Comment (Haadiiii) [reply src/pages/Categories.jsx:10]: yeah done 👍

**Review by Lucas-Masaba — APPROVED** [review]
# Hello @Haadiiii  👋 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

almost-there

## Highlights 🍾 
- Linter checks passing ✔️ 
- Added changes requested by the previous reviewer 👍 
- It is awesome that you used vite ⚡ 

## Optional suggestions 💭 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏 💻 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**